osutil/mount/mount_linux.go

```diff
@@ -0,0 +1,83 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package mount
+
+import (
+	"syscall"
+)
+
+// MountFlagsToOpts returns the symbolic representation of mount flags.
+func MountFlagsToOpts(flags int) (opts []string, unknown int) {
+	if f := syscall.MS_REMOUNT; flags&f == f {
```

Comment:
this code is still quite repetitive, what about something like this: 

```go
var flagStrToInt = map[string]int{
	"MS_REMOUNT":    syscall.MS_REMOUNT,
	"MS_BIND":       syscall.MS_BIND,
	"MS_REC":        syscall.MS_REC,
	"MS_RDONLY":     syscall.MS_RDONLY,
	"MS_SHARED":     syscall.MS_SHARED,
	"MS_SLAVE":      syscall.MS_SLAVE,
	"MS_PRIVATE":    syscall.MS_PRIVATE,
	"MS_UNBINDABLE": syscall.MS_UNBINDABLE,
}

// MountFlagsToOpts returns the symbolic representation of mount flags.
func MountFlagsToOpts(flags int) (opts []string, unknown int) {
	for flagStr, flagInt := range flagStrToInt {
		if f := flagInt; flags&f == f {
			flags ^= f
			opts = append(opts, flagStr)
		}
	}
	return opts, flags
}
```

Reply:
I'm not very keen to change this. The alternative proposal is involving dynamically allocated memory and has unpredictable ordering.

Reply:
Nitpicking, `append()` already allocates.  TBH, I'd prefer a list like I proposed elsewhere, but I'm not sure I want to bikeshed on this too much.